### PR TITLE
initial commit of GFS squash flags (do not merge)

### PR DESCRIPTION
**Pull request by shaneknapp**
initial commit of GFS squash configs for each deployment.

after setting up a bunch of GFS instances, i've discovered that it's much safer to have a deployment-specific config in each dir.

it's also important that the instance size in these files matches what's reported in gcloud/google cloud console.  if the number in here is lower, it will fail.  if it's higher, it will **quietly scale** the instance up!

https://github.com/berkeley-dsep-infra/datahub/issues/3898
https://github.com/berkeley-dsep-infra/datahub/pull/4002

### CODE REVIEW

**Comment by shaneknapp** [comment]
idea:  move these files to `config/filestore`, and stick a README.md in each dir w/the three-step instructions on how to deploy.  thoughts?

**Comment by ryanlovett** [comment]
I think the content of the READMEs can probably be moved to the docs somewhere, but it makes sense for the json to be in each deployments config.

I think it also makes sense to put the json into the cookiecutter template and to prompt for the storage size.

File: docs/admins/howto/new-hub.rst [hunk 1]
```diff
@@ -11,9 +11,8 @@ Why create a new hub?
 
 The major reasons for making a new hub are:
 
+#. A new course wants to join the Berkeley Datahub community!
 #. You want to use a different kind of authenticator.
-#. Some of your *students* are *admins* on another hub,
```
Comment (ryanlovett) [inline docs/admins/howto/new-hub.rst:15]: Isn't this still relevant?

**Review by ryanlovett — COMMENTED** [review]
This looks good to me, and thanks for adding the cookiecutter template and docs update!

Besides the doc change about student/admin access, just one other comment. I think the hub name doesn't need to be part of the squash-flags filename, e.g. it could just be `{hub}/config/filestore/squash-flags.json` for each, which is analogous to the various config/secret files like `common.yaml`, `prod.yaml`, `gcr-key.json`, etc. If you agree then the docs mentioning this file would need to be changed as well.

**Comment by balajialg** [comment]
Thanks for the docs update @shaneknapp!

**Comment by shaneknapp** [comment]
> This looks good to me, and thanks for adding the cookiecutter template and docs update!
> 
> Besides the doc change about student/admin access, just one other comment. I think the hub name doesn't need to be part of the squash-flags filename, e.g. it could just be `{hub}/config/filestore/squash-flags.json` for each, which is analogous to the various config/secret files like `common.yaml`, `prod.yaml`, `gcr-key.json`, etc. If you agree then the docs mentioning this file would need to be changed as well.

the primary driver for explicitly naming these files is to avoid confusion and accidentally upscaling storage when modifying `ROOT_SQUASH`.  granted, we probably won't be doing work en masse like this again (or for a long time), so it's probably fine to just name things `squash-flags.json`.

File: docs/admins/howto/new-hub.rst [hunk 2]
```diff
@@ -54,25 +64,40 @@ we provide that sets up a blank hub that can be customized.
...
+
+   .. code:: bash
+
+         drwxr-xr-x 4 ubuntu ubuntu     45 Nov  3 20:33 a11y-filestore
+	 drwxr-xr-x 4 ubuntu ubuntu     33 Jan  4  2022 astro-filestore
+	 drwxr-xr-x 4 ubuntu ubuntu  16384 Aug 16 18:45 biology-filestore
+
+   #. Now we need to set ``ROOT_SQUASH`` on the newly created mount. In the 
+      ``datahub/deployments/<hubname>/config/filestore/`` directory is a file named 
+      ``<hub-name>-squash-flags.json``. You will use this file when running the following ``gcloud``
```
Comment (ryanlovett) [inline docs/admins/howto/new-hub.rst]: With the latest changes you can remove `<hub-name>-` from `<hub-name>-squash-flags.json`.

File: docs/admins/howto/new-hub.rst [hunk 3]
```diff
@@ -54,25 +64,40 @@ we provide that sets up a blank hub that can be customized.
...
+	 drwxr-xr-x 4 ubuntu ubuntu  16384 Aug 16 18:45 biology-filestore
+
+   #. Now we need to set ``ROOT_SQUASH`` on the newly created mount. In the 
+      ``datahub/deployments/<hubname>/config/filestore/`` directory is a file named 
+      ``<hub-name>-squash-flags.json``. You will use this file when running the following ``gcloud`` 
+      command to apply the changes:
+
+   ..code:: bash
+
+     gcloud filestore instances update <filestore-instance-name> --zone=us-central1-b --flags-file=<hubname>-squash-flags.json
```
Comment (ryanlovett) [inline docs/admins/howto/new-hub.rst]: Similar to the one above, you can remove `<hubname>-` from `<hubname>-squash-flags.json`.

**Review by ryanlovett — APPROVED** [review]
LGTM! Just two doc changes which correspond to the filename change.

**Pull request merged**